docs/life-of-a-request.md

```diff
@@ -0,0 +1,200 @@
+# Life of a Request
+
+It can sometimes be confusing to figure out how to write a new feature,
+or debug an existing one. Let's try to follow a request through the
+Zulip codebase, and dive deep into how each part works.
+
+## A request is sent to the server, and handled by Nginx
+
+Everything in Zulip goes through nginx, when deployed in production.
+For the most part you don't need to know how this works, except for when
+it isn't working. Nginx does the first level of routing--deciding which
+application will serve the request (or deciding to serve the request
+itself for static content).
+
+## Nginx secures traffic with SSL
+
+If you visit your zulip server in your browser and discover that your
+traffic isn't being properly encrypted, an nginx configuration is the
+likely culprit.
+
+## Static files are served directly by Nginx
+
+Static files include javascript, css, static assets (like emoji, avatars),
+and user uploads (if stored locally and not on S3).
+
+```
+/home/zulip/prod-static
+```
+
+## Nginx routes other requests between tornado and django
+
+All our connected clients hold open long-polling connections so that
+they can recieve events (messages, presence notifications, and so on) in
+real-time. Events are served by Zulip's `tornado` application.
+
+Nearly every other kind of request is served by the `zerver` Django
+application.
+
+# Django routes the request to a view in urls.py files
+
+There are various urls.py files throughout the server codebase, and you
+can read about it more in [the directory structure doc](http://zulip.readthedocs.io/en/latest/directory-structure.html).
```

Comment:
My first thought was that the "it" here was intended for "various urls.py files" and was supposed to be a "them", but on rereading I assume the "it" is intended to match "the server codebase"? It might be worth just replacing "read about it more" with "read more about the codebase directory structure"